src/Runner.Listener/MessageListener.cs

```diff
@@ -32,18 +35,33 @@ public sealed class MessageListener : RunnerService, IMessageListener
         private ITerminal _term;
         private IRunnerServer _runnerServer;
         private TaskAgentSession _session;
+        private ICredentialManager _credMgr;
+        private IConfigurationStore _configStore;
         private TimeSpan _getNextMessageRetryInterval;
         private readonly TimeSpan _sessionCreationRetryInterval = TimeSpan.FromSeconds(30);
         private readonly TimeSpan _sessionConflictRetryLimit = TimeSpan.FromMinutes(4);
         private readonly TimeSpan _clockSkewRetryLimit = TimeSpan.FromMinutes(30);
         private readonly Dictionary<string, int> _sessionCreationExceptionTracker = new Dictionary<string, int>();
 
+        // Whether load credentials from .v2credentials file
+        internal bool _useV2Credentials;
+
+        // Whether .credentials existing, we will fallback to this if .v2credential doesn't work.
+        internal bool _v1CredentialsExists;
+
+        // need to check auth url if there is only .credentials and auth schema is OAuth
+        internal bool _needToCheckAuthorizationUrlUpdate;
+        internal Task<VssCredentials> _authorizationUrlMigrationBackgroundTask;
+        internal Task _authorizationUrlRollbackReattemptDelayBackgroundTask;
```

Comment:
when auth from Token service doesn't work, we will fallback to SPS and wait for 2 days before trying Token service again.